public/app/features/scenes/dashboard/DashboardsLoader.ts

```diff
@@ -129,6 +130,21 @@ export function createSceneObjectsForPanels(oldPanels: PanelModel[]): Array<Scen
           currentRowPanels = [];
         }
       }
+    } else if (panel.libraryPanel?.uid && !('model' in panel.libraryPanel)) {
+      const gridItem = new SceneGridItem({
+        body: new VizPanel({
+          title: panel.title,
```

Comment:
What do you think of implementing this instead via a LibraryPanel scene object that on activation loads the panel model from the backend and initializes a VizPanel (an inner body property). The LibraryPanel does have a component renderer that handles the loading state, and after loading just renders the VizPanel

This should make it easier to do the inverse transform back to the persisted model

Reply:
Yeah, that would make sense, haven't thought about that! Will do it - it will definitely help serialization.

Reply:
this would also make the loading of the library panel lazy (for a lazy loaded grid layout)

Reply:
this would also make the loading of the library panel lazy (for a lazy loaded grid layout)